cachi2/core/models/sbom.py

```diff
@@ -284,6 +316,43 @@ def __hash__(self) -> int:
         return hash((self.annotator, self.annotationDate, self.annotationType, self.comment))
 
 
+class SPDXChecksum(pydantic.BaseModel):
+    """A basic representation of a checksum entry."""
+
+    algorithm: str
+    checksumValue: str
+
+    def __hash__(self) -> int:
+        return hash(self.algorithm + self.checksumValue)
```

Comment:
Is this to make instances hashable and therefore usable in maps/sets? I suppose so, but this is insufficient because the instances are still mutable, a single update to any of the values invalidates all places that such an instance would have been used as a hashable object. I know it may be unlikely, but I feel like there are enough models here where it might be quite hard to track down a potential violator. Can we make use of the `frozen` config option for these models?
https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.frozen